flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/slotmanager/DefaultResourceTrackerTest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.resourcemanager.slotmanager;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.clusterframework.types.ResourceProfile;
+import org.apache.flink.runtime.slots.ResourceRequirement;
+import org.apache.flink.util.TestLogger;
+
+import org.hamcrest.collection.IsMapContaining;
+import org.junit.Test;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.empty;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Tests for the {@link DefaultResourceTracker}.
+ *
+ * <p>Note: The majority is of the tracking logic is covered by the {@link JobScopedResourceTrackerTest}.
+ */
+public class DefaultResourceTrackerTest extends TestLogger {
+
+	@Test
+	public void testInitialBehavior() {
+		DefaultResourceTracker tracker = new DefaultResourceTracker();
+
+		assertThat(tracker.isEmpty(), is(true));
+		tracker.notifyLostResource(JobID.generate(), ResourceProfile.ANY);
+		tracker.clear();
```

Comment:
`clear()` is used during the suspension of the slot manager (which is kinda annoying in general).

Reply:
I am a bit confused because we don't assert anything after these lines.

If it tests that no exceptions are thrown, should we add a separate test case with a name along the lines of `clearDoesNotThrowException`?